service/history/execution/context.go

```diff
@@ -852,6 +858,10 @@ func (c *contextImpl) UpdateWorkflowExecutionWithNew(
 		domainName,
 		resp.MutableStateUpdateSessionStats,
 	)
+	c.stats.BlobSize = currentWorkflowSize - oldWorkflowSize
+	c.stats.OldHistoryCount = oldWorkflowHistoryCount
+	c.stats.OldHistorySize = oldWorkflowSize
```

Comment:
huh.  `Set/GetHistorySize` just forwards to c.stats like this.
tbh I think I like the bare field access better than the func, unless something external needs it (e.g. as an interface).  so 👍 maybe it's even worth removing those wrappers around HistorySize.